google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/MessageDispatcher.java

```diff
@@ -103,7 +109,7 @@ void extendExpiration(Instant now) {
       expiration = now.plus(Duration.standardSeconds(nextExtensionSeconds));
```

Comment:
I think there is still a problem. Imagine the job is created at time 0 and maxAckExtensionPeriod is ten minutes. If the message is not acked or nacked, we continue to modify the ack deadline an increasing amount each time. As a result, we could overshoot the maxAckExtensionPeriod. I believe we could, for example, end up extending an additional 5 minutes at minute 8, which means we would not stop modifying the ack deadline until minute 13, which is 3 minutes beyond maxAckExtensionPeriod. That's why I think you need to set expiration to min(now.plus(Duration.standardSeconds(nextExtensionSeconds)), creation.plus(maxAckExtensionPeriod)).